src/metabase/driver.clj

```diff
@@ -317,7 +317,8 @@
   :hierarchy #'hierarchy)
 
 (defmulti describe-table-indexes
-  "Returns a set of field names that are either indexed or are the first columns in a composite index."
+  "Returns a set of field names that are either indexed or are the first columns in a composite index.
+  Field name can be a string, nested field need to be represented as a sequence of strings."
```

Comment:
Thinking ahead to being able to supporting composite indices, I don't love that nested fields are represented as a sequence of strings. It's going to make change harder and more error prone.

Perhaps this function should return a set of maps? e.g.
`{:type :normal-column-index, :field-name "..."}`
`{:type :nested-column-index, :nfc-path ["..." "..."]}`

Reply:
yea makes sense.

One thing I'd change is to use a map of 2 keys `[:type :value]`.

The type is enough to differentiate the field name / path is represented, we don't need to use a different name for it.

Implement it here https://github.com/metabase/metabase/pull/36624/commits/20055e1504e1b24dc830ac3bb4ee1e0b823cd7f4